test/framework/seedregistrationframework.go

```diff
@@ -0,0 +1,383 @@
+	flag.StringVar(&newCfg.Provider, "provider-region", "", "provider region e.g. eu-west-1")
+	//Network
+	flag.StringVar(&newCfg.NetworkingDefServices, "networking-default-shoot-services", "", "Value for Seed.Spec.Networks.ShootDefaults.Services")
+	flag.StringVar(&newCfg.NetworingDefPods, "networking-default-shoot-pods", "", "Value for Seed.Spec.Networks.ShootDefaults.Pods")
+	flag.StringVar(&newCfg.NetworingkBlockCIDRs, "networking-blockcidrs", "", "Comma-separated values for seed.Spec.Networks.BlockCIDRs.")
+	flag.StringVar(&newCfg.NetworingPods, "networking-pods", "", "Value for Seed.Spec.Networks.Pods")
+	flag.StringVar(&newCfg.NetworingServices, "networking-services", "", "Value for Seed.Spec.Networks.Services")
+	flag.StringVar(&newCfg.NetworingkNodes, "networking-nodes", "", "Value for Seed.Spec.Networks.Nodes")
+	flag.StringVar(&newCfg.DeployGardenletOnShoot, "deploy-gardenlet", "", "Specify if a gardanlet instance should be installed on the shoot. Default = true")
+
+	seedRegistrationConfig = newCfg
+
+	return seedRegistrationConfig
+}
+
+//AddSeed function tries to get an existing seed using the configured GardenClient
+//and the given seed parameter and adds it to the SeedRegistrationFramework
+func (f *SeedRegistrationFramework) AddSeed(ctx context.Context, seed *gardencorev1beta1.Seed) error {
+	if f.GardenClient == nil {
+		return errors.New("no gardener client is defined")
+	}
+	if err := f.GardenClient.DirectClient().Get(ctx, client.ObjectKey{Name: seed.Name}, seed); err != nil {
+		return err
+	}
+	f.Seed = seed
+	return nil
+}
+
+//RegisterSeed registers a new seed and its secret reference (seed.Spec.SecretRef)
+//according to the configuration provided for the SeedRegistrationFramework
+func (f *SeedRegistrationFramework) RegisterSeed(ctx context.Context) (err error) {
+	if f.GardenClient == nil {
+		return errors.New("no gardener client is defined")
+	}
+
+	refShoot := &gardencorev1beta1.Shoot{}
+	if err = f.GardenClient.DirectClient().Get(ctx, kutil.Key(f.Config.ShootedSeedNamespace, f.Config.ShootedSeedName), refShoot); err != nil {
+		return err
+	}
+
+	if deploy, _ := strconv.ParseBool(f.Config.DeployGardenletOnShoot); deploy {
+		return f.deployGardenletOnShoot(ctx, refShoot)
+	}
+
+	secretBinding := &gardencorev1beta1.SecretBinding{}
+	if err = f.GardenClient.DirectClient().Get(ctx, kutil.Key(f.Config.ShootedSeedNamespace, f.Config.SecretBinding), secretBinding); err != nil {
+		return err
+	}
+
+	f.Seed = f.buildSeedObject(*secretBinding, *refShoot)
+
+	if err = f.generateSeedSecret(ctx, secretBinding); err != nil {
+		f.Logger.Fatalf("Cannot generate seed secret %s", err)
+		return err
+	}
+	f.Logger.Infof("Creating seed secret %s in namespace %s", f.Secret.GetName(), f.Secret.GetNamespace())
+	//Apply secret to the cluster
+	if err = client.IgnoreNotFound(f.GardenClient.DirectClient().Create(ctx, f.Secret)); err != nil {
+		f.Logger.Fatalf("Cannot create seed secret %s: %s", f.Secret.GetName(), err)
```

Comment:
As far as I see Fatalf will call os.Exit(1). Is it really what we want or we want to return err to the caller?